cache/in-memory/src/event/guild.rs

```diff
@@ -118,6 +118,10 @@ impl InMemoryCache {
             self.cache_stage_instances(id, stage_instances);
         }
 
+        if !self.wants(ResourceType::GUILD) {
+            return;
+        }
+
```

Comment:
I'd personally change this to
```rust
if self.wants(ResourceType::GUILD) {
    let guild = CachedGuild { .. };
    self.unavailable_guilds.remove(&guild.id());
    self.guilds.insert(guild.id(), guild);
```

Reply:
Ye I had this at first too, but then I thought the other functions do it with a return, so I did that here too.

Reply:
i think we can switch to what vilgotf suggested